src/algorithm/mod.rs

```diff
@@ -31,9 +32,9 @@ pub enum ButterflyEnum<T> {
 	Butterfly2(butterflies::Butterfly2),
 	Butterfly3(butterflies::Butterfly3<T>),
 	Butterfly4(butterflies::Butterfly4),
-	Butterfly5(butterflies::Butterfly5<T>),
+	Butterfly5(Box<butterflies::Butterfly5<T>>),
```

Comment:
Yes. If you look at the size of the MixedRadixDoubleButterfly, right now it's over 100 bytes for f32, and over 200 bytes for f64, and almost all of it comes from the 6-element array in the butterfly7 struct.

For most FFTs this will be purely wasted space. I think we have to be ok with some wasted space, but 5 and 7 are the two worst for space.